lib/diagnostics_channel.js

```diff
@@ -105,27 +188,17 @@ function channel(name) {
   }
 
   channel = new Channel(name);
-  channels[name] = new WeakReference(channel);
+  channel._weak = new WeakReference(channel);
+  channels[name] = channel._weak;
```

Comment:
Probably? I'll have to make sure the GC characteristics are still correct that way. I _think_ it should be fine though.

Reply:
There should be no difference. In the end you add two properties to the new `channel` instance. Not sure why we need two properties referring to the same `WeakReference`.

Reply:
Ah, don't actually need `_weak` anymore. That's leftover from something I had before to hold channels open _through_ `TracingChannel` as I was constructing them directly before, but I changed how that works so that's not relevant anymore.

Reply:
Just noticed that this is not resulting in two properties on `channel` as it is `channels[name]`. I guess that is still needed to allow a setup where a consumer subscribes on a channel but doesn't keep a reference to the channel instance.

Reply:
It can go back to how it was before where the `WeakReference` is stores directly in `channels[name]` rather than _also_ needing to exist on the channel itself. That was just there because I had some weird weak reference forwarding thing for TracingChannel which turned out to be a bad idea so I deleted it. 😅